IPython/core/display.py

```diff
@@ -607,9 +620,19 @@ def data(self, svg):
             pass
         svg = cast_unicode(svg)
         self._data = svg
+    
+    def _data_and_metadata(self):
+        """shortcut for returning metadata with shape information, if defined"""
+        md = {}
+        if self.metadata:
+            md.update(self.metadata)
```

Comment:
I think it would be equivalent to deepcopy(self.metadata) not self.metadata.copy() (technically). I'll change that.